src/discussion/discussion.service.ts

```diff
@@ -0,0 +1,139 @@
+            
+        } catch (error) {
+            throw new BadRequestException(
+                ZaLaResponse.BadRequest(error.name, error.message, error.status)
+            )
+        }
+    }
+
+    async addChildComment(profileId: string, parentCommentId: string, dto: CreateChildCommentDto): Promise<Comment>{
+        try {
+            //check if comment is a parent comment
+            const isParentComment = await this.commentRepo.findOne({where:{id: parentCommentId}})
+            if(isParentComment.is_parent != true){
+                throw new BadRequestException(
+                    ZaLaResponse.BadRequest('You cannot only add a comment to a parent comment', '400')
+                )
+            }
+
+            const childCommentObj = await this.commentRepo.create({
+                ...dto,
+                profile_id: profileId   
+            })
+            
+            const childComment = await this.commentRepo.save(childCommentObj) 
+
+            //update parent comment
+            isParentComment.child_comment_ids.push(childComment.id)
+            await this.commentRepo.save(isParentComment)
+            
+            return childComment
+
+        } catch (error) {
+            throw new BadRequestException(
+                ZaLaResponse.BadRequest(error.name, error.message, error.status)
+            )
+        }
+    }
+
+    async editComment(profileId: string, commentId: string, dto: UpdateCommentDto): Promise<void>{
+        try {
+            const comment = await this.commentRepo.findOne({where:{id: commentId}})
+            if(profileId !== comment.profile_id){
+                throw new BadRequestException(
+                    ZaLaResponse.BadRequest(
+                        'Unauthorized request',
+                        'You can only edit a comment that belongs to you', 
+                        '401')
+                )
+            }
+            await this.commentRepo.update(commentId, dto)
+            
+        } catch (error) {
+            throw new BadRequestException(
+                ZaLaResponse.BadRequest(error.name, error.message, error.status)
+            )
+        }
+    }
+
+    async getParentAndChildComments(parentCommentId: string):Promise<Comment[]>{
+        try {
+            const comments = []
+            const parentComment = await this.commentRepo.findOne({where:{id:parentCommentId}})
+            comments.push(parentComment)
+
+            const childComments = parentComment.child_comment_ids
+
+            for(let comment = 0; comment <= childComments.length - 1; comment++){
+                const childComment = await this.commentRepo.findOne({where:{id: childComments[comment]}})
+                comments.push(childComment)
+            }
+            
+            return comments
+        } catch (error) {
+            throw new BadRequestException(
+                ZaLaResponse.BadRequest(error.name, error.message, error.status)
+            )   
+        }
+    }
```

Comment:
I don't think this is efficient, if a comment should have comments under it, it should be another entity

Reply:
Brain suggested it while we were discussing it. I'll bring it up to him during the standup this morning